modules/tinymce/src/plugins/link/test/ts/browser/LinkContextMenuTest.ts

```diff
@@ -18,14 +18,32 @@ describe('browser.tinymce.plugins.link.LinkContextMenuTest', () => {
 
   const pressDownArrowKey = (editor: Editor) => TinyUiActions.keydown(editor, Keys.down());
 
-  it('TINY-2293: Context menu shows up on links', async () => {
-    const editor = hook.editor();
-    editor.setContent('<p><a href="http://tiny.cloud">tiny</a></p>');
-    await TinyUiActions.pTriggerContextMenu(editor, 'a[href="http://tiny.cloud"]', '.tox-silver-sink [role="menuitem"]');
+  const pTestContextMenuItems = async (editor: Editor) => {
     await pAssertFocusOnItem('Link...', '.tox-collection__item:contains("Link...")');
     pressDownArrowKey(editor);
     await pAssertFocusOnItem('Remove link', '.tox-collection__item:contains("Remove link")');
     pressDownArrowKey(editor);
     await pAssertFocusOnItem('Open link', '.tox-collection__item:contains("Open link")');
+  };
+
+  it('TINY-2293: Context menu shows up on links', async () => {
+    const editor = hook.editor();
+    editor.setContent('<p><a href="http://tiny.cloud">tiny</a></p>');
+    await TinyUiActions.pTriggerContextMenu(editor, 'a[href="http://tiny.cloud"]', '.tox-silver-sink [role="menuitem"]');
+    await pTestContextMenuItems(editor);
+  });
+
+  it('TINY-7993: Context menu shows correctly right after inserting a link', async () => {
+    const editor = hook.editor();
+    editor.setContent('<p>aaa bbb ccc</p>');
+    TinySelections.setSelection(editor, [ 0, 0 ], 'aaa '.length, [ 0, 0 ], 'aaa bbb'.length);
+    editor.execCommand('mceLink');
+    await TinyUiActions.pWaitForDialog(editor);
+    FocusTools.setActiveValue(SugarDocument.getDocument(), 'http://tiny.cloud');
+    TinyUiActions.submitDialog(editor);
+    await Waiter.pTryUntil('Wait for content to change', () =>
+      TinyAssertions.assertContent(editor, '<p>aaa <a href="http://tiny.cloud">bbb</a> ccc</p>'));
+    await TinyUiActions.pTriggerContextMenu(editor, 'a[href="http://tiny.cloud"]', '.tox-silver-sink [role="menuitem"]');
+    await pTestContextMenuItems(editor);
```

Comment:
also, you need to assure that `Open Link` context menu item is not disabled

Reply:
You are right, thanks! Added the test [here](https://github.com/tinymce/tinymce/pull/8386/commits/1dffd34dde59c4fd60630aadb5dbcbdb303ed13d).